proxy/envoy/discovery.go

```diff
@@ -526,9 +545,14 @@ func (ds *DiscoveryService) ListRoutes(request *restful.Request, response *restf
 			return
 		}
 
-		instances := ds.Discovery.HostInstances(map[string]bool{ip: true})
-		services := ds.Discovery.Services()
-		httpRouteConfigs := buildOutboundHTTPRoutes(instances, services, ds.Accounts, ds.MeshConfig, ds.Config)
+		var httpRouteConfigs HTTPRouteConfigs
+		if node == ingressNode {
+			httpRouteConfigs, _ = buildIngressRoutes(ds.Config.IngressRules(""))
+		} else {
+			instances := ds.Discovery.HostInstances(map[string]bool{node: true})
+			services := ds.Discovery.Services()
+			httpRouteConfigs = buildOutboundHTTPRoutes(instances, services, ds.Accounts, ds.MeshConfig, ds.Config)
+		}
```

Comment:
Bit unclear on the logic here. Will the envoy config in ingress container use CDS instead of hard coded clusters? (This allows us to set circuit breakers, change load balancing algorithms, etc). 

Will the envoy config in ingress container use RDS or use the spec from ingress resource (which just allows path, prefix, virtual hosts) or merge both based on the backend service selected in ingress? The latter would be the best of both, as we could define route rules (header match, retry, timeout, weighted clusters) for the destination